src/hera/task.py

```diff
@@ -819,6 +819,8 @@ def _build_template(self) -> Optional[IoArgoprojWorkflowV1alpha1Template]:
 
         if self.source is not None:
             setattr(template, "script", self._build_script())
+        elif self.resource_template is not None:
```

Comment:
Thank you for the fix @jalkjaer! Any chance this PR can include a test for this? 🙂

Reply:
Done